front/cls-wpsd-front.php

```diff
@@ -1,780 +1,858 @@
+                return $wpsdGeneralSettings[$key];
+            }
+        }
+        return $default;
+    }
+
+    public function wpsd_load_shortcode_view($atts)
+    {
+        $output = '';
+
+        wp_localize_script($this->wpsd_assets_prefix . 'front-script', 'wpsdSetShortcodes', $atts);
+
+        // temporary...until I figure out how to call the function above...
+        $default_donation_amounts = [10, 20, 50, 100];
+        $wpsdGeneralSettings = stripslashes_deep(unserialize(get_option('wpsd_general_settings')));
+        if (is_array($wpsdGeneralSettings)) {
+            if (array_key_exists('wpsd_donation_amounts', $wpsdGeneralSettings) && !empty($wpsdGeneralSettings['wpsd_donation_amounts'])) {
+                $default_donation_amounts = $wpsdGeneralSettings['wpsd_donation_amounts'];
+            }
+        }
+
+        // donation amounts to int values
+        if (array_key_exists('donation_amounts', $atts)) {
+            $atts['donation_amounts'] = array_map('intval', explode(',', $atts['donation_amounts']));
+        }
+
+        ob_start();
+
+        $expected_attr = array(
+            "campaign" => "General",
+            'campaign_id' => "",
+            'custom_amount' => "true",
+            'allow_recurring' => "true",
+            "fund" => "General",
+            "fund_id" => "",
+            "imof" => "",
+            "donation_amounts" => $default_donation_amounts,
+        );
+
+        $params = shortcode_atts($expected_attr, $atts);
+
+
+        include(plugin_dir_path(__FILE__) . '/view/wpsd-front-view.php');
+        $output .= ob_get_clean();
+        return $output;
+    }
+
+    function wpsd_get_states_handler()
+    {
+        if (!isset($_GET['code']) || empty($_GET['code'])) {
+            $err = array(
+                "status" => "error",
+                "message" => esc_html__('Country Code is required', 'wp-stripe-donation'),
+            );
+            wp_send_json_error($err, 400);
+        }
+        $country_code = $_GET['code'];
+        /** @var  \MenaraSolutions\Geographer\Country $country */
+        $country = $this->countries->findOne(array('code' => $country_code));
+        if (!$country) {
+            $err = array(
+                "status" => "error",
+                "message" => esc_html__('No country found', 'wp-stripe-donation'),
+            );
+            wp_send_json_error($err, 400);
+        }
+        $states = $country->getStates()->sortBy("name")->toArray();
+        $data = array(
+            "status" => "success",
+            "states" => $states,
+        );
+        wp_send_json_success($data);
+    }
+
+    function wpsd_donation_handler()
+    {
+        $payload = @file_get_contents('php://input');
+        $data = json_decode($payload, true);
+        // @ts 2023-03-01 22:40
+        error_log(__METHOD__ . ' +' . __LINE__ . ' $data: ' . var_export($data, true));
```

Comment:
@joelcrawford I would clean up such debug log in a next commit and I will create a pull request for you